### PR TITLE
Feature - Integrate test in ci flow for Validator POS Node

### PR DESCRIPTION
**Pull request by huythanhcere**
Set up tests execution during CI for Validator PoS node for PRs. Jira ticket ([url](https://cerenetwork.atlassian.net/browse/CBI-81?atlOrigin=eyJpIjoiMTBjZDJlNjI3NzY0NDRiNzk2NjMxYmVjZjZiMmE3YTAiLCJwIjoiaiJ9)).

- Add test command execution in docker build step

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,8 @@ RUN apt-get update && \
 RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
          export PATH=$PATH:$HOME/.cargo/bin && \
          scripts/init.sh && \
-         cargo +nightly-2020-10-06 build --$PROFILE
+         cargo test && \
+         cargo build --$PROFILE
```
Comment (AndreiNavoichyk) [inline Dockerfile]: @huythanhcere 
Move back `+nightly-2020-10-06`. We will do migration separately.

**Review by AndreiNavoichyk — COMMENTED** [review]
How is it going to be applied in PR?

File: .github/workflows/build-and-push-docker-image.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ on:
   push:
     branches:
       - 'dev-cere'
+      - 'feature/integrate-test-in-ci-flow'
 env:
```
Comment (AndreiNavoichyk) [inline .github/workflows/build-and-push-docker-image.yml]: @huythanhcere 
You should use `on_pull_request` event. Please, read about it [here](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#configuring-workflow-events).

**Review by AndreiNavoichyk — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-push-docker-image.yml (hunk 2 above)
Comment (huythanhcere) [reply .github/workflows/build-and-push-docker-image.yml]: Ok, i will change to use on_pull_request event in the next commit.

File: Dockerfile (hunk 1 above)
Comment (huythanhcere) [reply Dockerfile]: After tried with "+nightly-2020-10-06", i tried to remove it. This change is temporary. I will move back in the next commit.

File: .github/workflows/build-and-push-docker-image.yml (hunk 2 above)
Comment (AndreiNavoichyk) [reply .github/workflows/build-and-push-docker-image.yml]: Also, `on_pull_request` only tests should be executed.

Build should be on `push`

File: .github/workflows/test-image.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  IMAGE_TAG: ${{ github.run_number }}-${{ github.sha }}
+jobs:
+  build-and-push:
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: 'Checkout repository'
+        uses: 'actions/checkout@v1'
+      - name: 'Checkout submodules'
+        run: 'git submodule update --init --recursive'
+      - name: 'Configure AWS credentials'
```
Comment (AndreiNavoichyk) [inline .github/workflows/test-image.yml:21]: We don't need to push smth so we don't need configure AWS credentials

File: .github/workflows/test-image.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        uses: 'aws-actions/configure-aws-credentials@v1'
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: 'us-west-2'
+      - name: 'Login to Amazon ECR'
```
Comment (AndreiNavoichyk) [inline .github/workflows/test-image.yml:27]: We don't need this section.

File: .github/workflows/test-image.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        id: 'login-ecr'
+        uses: 'aws-actions/amazon-ecr-login@v1'
+      - name: 'Build and push image'
```
Comment (AndreiNavoichyk) [inline .github/workflows/test-image.yml:30]: We don't need push, rename. Only testing.

File: Dockerfile.UT [hunk 6]
```diff
@@ -0,0 +1,14 @@
+FROM phusion/baseimage:0.11 as builder
+LABEL maintainer="team@cere.network"
+LABEL description="This is the build stage to create the binary."
+ARG PROFILE=release
+WORKDIR /cerenetwork
+COPY . /cerenetwork
+
+RUN apt-get update && \
+	apt-get upgrade -y && \
```
Comment (AndreiNavoichyk) [inline Dockerfile.UT:9]: Align spaces. As an example take a look to [this](https://github.com/Cerebellum-Network/pos-network-node/blob/dev-cere/Dockerfile#L8).

File: .github/workflows/test-image.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
+name: 'Automation test flow'
+on:
+  push:
+    branches:
+      - 'dev-cere'
+      - 'feature/integrate-test-in-ci-flow'
```
Comment (AndreiNavoichyk) [inline .github/workflows/test-image.yml:6]: Remove this line.

File: .github/workflows/test-image.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  pull_request:
+    branches:
+      - 'dev-cere'
```
Comment (AndreiNavoichyk) [inline .github/workflows/test-image.yml:9]: Here should be any branch. Use wildcards.

File: .github/workflows/test-image.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+env:
+  ECR_REPOSITORY: 'crb-substrate-testnet-node'
```
Comment (AndreiNavoichyk) [inline .github/workflows/test-image.yml:11]: We don't need it, clean up.

**Comment by huythanhcere** [comment]
Doing PR, close because of new PR are created with only result commit.

**Pull request closed without merging**